client/ayon_core/hosts/houdini/plugins/publish/extract_active_view_thumbnail.py

```diff
@@ -0,0 +1,48 @@
+import pyblish.api
+import tempfile
+from ayon_core.pipeline import publish
+from ayon_core.hosts.houdini.api import lib
+from ayon_core.hosts.houdini.api.pipeline import IS_HEADLESS
+
+
+class ExtractActiveViewThumbnail(publish.Extractor):
+    """Set instance thumbnail to a screengrab of current active viewport.
+
+    This makes it so that if an instance does not have a thumbnail set yet that
+    it will get a thumbnail of the currently active view at the time of
+    publishing as a fallback.
+
+    """
+    order = pyblish.api.ExtractorOrder + 0.49
+    label = "Extract Active View Thumbnail"
+    families = ["workfile"]
+    hosts = ["houdini"]
+
+    def process(self, instance):
+        if IS_HEADLESS:
+            self.log.debug(
+                "Skip extraction of active view thumbnail, due to being in"
+                "headless mode."
+            )
+            return
+
+        thumbnail = instance.data.get("thumbnailPath")
+        if not thumbnail:
+            view_thumbnail = self.get_view_thumbnail(instance)
+            if not view_thumbnail:
+                return
+
+            self.log.debug("Setting instance thumbnail path to: {}".format(
+                view_thumbnail
+            ))
+            instance.data["thumbnailPath"] = view_thumbnail
+
+    def get_view_thumbnail(self, instance):
+
+        sceneview = lib.get_scene_viewer()
```

Comment:
There may be cases where `get_scene_viewer` returns `None`. Instead of actually creating the temporary file that might be a good case to capture early since it'd mean the snapshot will fail and return None anyway.

So:
```python
if sceneview is None:
    return
```

Do nothing - maybe a log a warning that it's being skipped because no scene viewer was found or whatever.